lib/src/model/uuid.dart

```diff
@@ -13,27 +13,31 @@ class Uuid {
 
     var byteOffset = 0;
     for (var substringStart = 0; substringStart < string.length;) {
-      if (byteOffset >= 16) {
-        throw _UuidParseFailure(string);
-      }
-
       if (string[substringStart] == "-") {
         substringStart += 1;
         continue;
       }
 
+      if (byteOffset >= 16 || substringStart + 2 > string.length) {
+        throw _UuidParseFailure(string);
+      }
+
       final byte = int.tryParse(
-          string.substring(substringStart, substringStart + 2),
-          radix: 16);
+        string.substring(substringStart, substringStart + 2),
+        radix: 16,
+      );
       if (byte == null) throw _UuidParseFailure(string);
 
       data[byteOffset] = byte;
 
       byteOffset += 1;
       substringStart += 2;
     }
-
-    return Uuid(data.buffer.asUint8List(0, byteOffset));
+    if (byteOffset == 2 || byteOffset == 16) {
```

Comment:
Only 2 bytes ([registered](https://www.bluetooth.com/specifications/gatt/services/) short service IDs) and 16 bytes (everything else) (UU)IDs are used in BLE.

Reply:
Correction, according to [CBUUID](https://developer.apple.com/documentation/corebluetooth/cbuuid) docs, 2/4/16 bytes are used. Will change this and the unit-tests.